plc4j/spi/src/main/java/org/apache/plc4x/java/spi/Plc4xNettyWrapper.java

```diff
@@ -152,6 +152,11 @@ protected void decode(ChannelHandlerContext channelHandlerContext, T t, List<Obj
                 continue;
             }
             // Timeout?
+            if (registration.isDone()) {
```

Comment:
There is earlier condition which indicates that registration is cancelled. Maybe making a switch to `isActive` or similar would let us merge both? From `Plc4xNettyWrapper` perspective it doesn't matter if handler is cancelled or timed out, its relevant if its still needed to be looped over. With changes you did I think we got cancel and done callbacks within registration where appropriate log statements can be made.

Reply:
Yes, I think you are right, these two could be merged together. Moreover, if `isCancelled()` is `true`, `isDone()` is also `true`, so it would be already enough to check the latter.